ajax/index.html

```diff
@@ -251,102 +252,121 @@ <h1 class="center">Asynchronous Technologies: AJAX/JSON</h1>
+  id: 'ab92ed',
+  fluffy: true
 }
+
+data = JSON.stringify( data ); // transform to string
+
+request.open('post', '/save/the/kittens', true);
+request.setRequestHeader('Content-type', 'application/json');
+request.send(data); // send request with data
 </pre>
-          <p>You can open either a GET or POST connection to a URL, specify a callback function, and then send the request. </p>
         </section>
-      </div><!--sending an XMLHttpRequest-->
+      </div><!--XMLHttpRequest with POST-->
 
       <div class="slide normal">
-        <header><h1>3) Handling the Response</h1></header>
+        <header><h1>XMLHttpRequest error handling</h1></header>
         <section class="content">
 <pre>
-function processRequest() { 
-  if (request.readyState == 4) { 
-    if(request.status == 200) { 
-      processResponse(request);
-    } else { 
-     alert("Error loading page " +
-       request.status + ":" +
-       request.statusText); 
-    } 
-  } 
+var request = new XMLHttpRequest();
+
+request.onload = function() {
+  // use the .status property to check the status code
+  if ( request.status === 200 ) {
+    success( JSON.parse( request.responseText ) );
+  } else {
+    console.log( 'Error!' )
+  }
+}
+
+// handle errors
+request.onerror = function() {
+  console.log( 'Error!' )
 }
+  
+request.open('get', '/path/to/api', true);
+request.setRequestHeader('Content-type', 'application/json');
+request.send();
 </pre>
-          <p>You can use the <strong>readyState</strong> property to check if the request has been completed, and status to check if it completed successfully</p>
         </section>
-      </div><!--handling the response-->
+      </div><!--XMLHttpRequest error handling-->
 
       <div class="slide normal">
-        <header><h1>4) Processing the Response</h1></header>
+        <header><h1>XMLHttpRequest response handling</h1></header>
         <section class="content">
-          <p>books.json:</p>
 <pre>
-[
- {"title": "Robot Dreams",
-  "author": "Isaac Asimov",
-  "categories": ["scifi", "shortstories"]
-  },
-  {"title": "Pirate Cinema",
-   "author": "Cory Doctorow",
-   "categories": ["scifi", "romance"]
-  }
-]
+request.onload = function() {
+  processReponse( JSON.parse( request.responseText ) );
+}
+
+function processResponse(data) {
+  var books = data; 
+
+  for (var i = 0; i < books.length; i++) { 
+      var book = books[i];
+      var p = document.createElement('p');
+      p.innerHTML = book.title + ' by ' + book.author;
+      document.body.appendChild(p);
+    }
+} 
 </pre>
+        <p>Would evaluate to:<br>
+          Robot Dreams, by Isaac Asimov<br>
+          Pirate Cinema, by Cory Doctorow
+        <p>
         </section>
-      </div><!--processing the response-->
+      </div><!--XMLHttpRequest response handling-->
 
       <div class="slide normal">
-        <header><h1>4) Processing the Response</h1></header>
+        <header><h1>XMLHttpRequest in IE</h1></header>
```

Comment:
Yes... I was recently asked about support for this and fallbacks... it's worth a mention on an earlier slide.